ecs_client_test.go

```diff
@@ -19,11 +19,32 @@ import (
 func TestECSClientInterface(t *testing.T) {
 	assert.Implements(t, (*ECSClient)(nil), &BasicECSClient{})
 }
+func TestECSClientTaskDefinition(t *testing.T) {
+
+	cleanupTaskDefinition := func(ctx context.Context, t *testing.T, c *BasicECSClient, out *ecs.RegisterTaskDefinitionOutput) {
+		if out != nil && out.TaskDefinition != nil && out.TaskDefinition.TaskDefinitionArn != nil {
+			out, err := c.DeregisterTaskDefinition(ctx, &ecs.DeregisterTaskDefinitionInput{
+				TaskDefinition: out.TaskDefinition.TaskDefinitionArn,
+			})
+			require.NoError(t, err)
+			require.NotZero(t, out)
+		}
+	}
+
+	cleanupTask := func(ctx context.Context, t *testing.T, c *BasicECSClient, runOut *ecs.RunTaskOutput) {
+		out, err := c.StopTask(ctx, &ecs.StopTaskInput{
```

Comment:
This is making some assumptions on the output that could panic. Can we check that all the inputs are non-nil?
```go
runOut != nil && `len(runOut.Tasks) > 0 && runOut.Tasks[0].TaskArn != nil
```